# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: c0a4aecf98b0
# Full Hash: c0a4aecf98b026eb7f88e5d105f85c3572ed522a
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 1 - Allow calling BrowserParent::InitRendering multiple times, and remove RenderFrame dependency from nsFrameLoader. r=kats
#   
#   This cleanup will simplify refactoring nsFrameLoader later.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31429
# ==============================================================================

diff -r c8e9b6a81194 -r c0a4aecf98b0 dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Mon May 20 10:01:02 2019 -0700
+++ b/dom/base/nsFrameLoader.cpp	Wed Apr 24 22:36:27 2019 -0500
@@ -1063,15 +1063,10 @@
       return true;
     }
 
-    RenderFrame* rf =
-        mBrowserParent ? mBrowserParent->GetRenderFrame() : nullptr;
-
-    if (!rf || !rf->AttachLayerManager()) {
-      // This is just not going to work.
+    if (!mBrowserParent->Show(
+            size, ParentWindowIsActive(mOwnerContent->OwnerDoc()))) {
       return false;
     }
-
-    mBrowserParent->Show(size, ParentWindowIsActive(mOwnerContent->OwnerDoc()));
     mRemoteBrowserShown = true;
 
     nsCOMPtr<nsIObserverService> os = services::GetObserverService();